### PR TITLE
[deprecation round2] LPS-104401 portal-kernel part6

### PR DESCRIPTION
**Pull request by ChiZelin**

### CODE REVIEW

**Comment by ChiZelin** [comment]
ci:test:sf

**Comment by ChiZelin** [comment]
ci:test

**Comment by dantewang** [comment]
Fetched

File: portal-impl/test/unit/com/liferay/portal/json/JSONSerializerTest.java [hunk 1]
```diff
@@ -85,33 +86,49 @@ public void testSerializeHits() {
 	public void testSerializeServiceContext() {
 		ServiceContext serviceContext = new ServiceContext();
 
+		serviceContext.setModelPermissions(
+			ModelPermissionsFactory.createWithDefaultPermissions(null));
```
Comment (dantewang) [inline portal-impl/test/unit/com/liferay/portal/json/JSONSerializerTest.java]: @ChiZelin `ModelPermissionsFactory` depends on various service builder services which are not available in unit test. I think you should switch back to directly use `ModelPermissions`, just like what the methods you removed from `ServiceContext` does.

File: modules/dxp/apps/portal-reports-engine-console/portal-reports-engine-console-service/src/main/java/com/liferay/portal/reports/engine/console/service/impl/DefinitionLocalServiceImpl.java [hunk 2]
```diff
@@ -209,12 +211,18 @@ public Definition updateDefinition(
 
 		// Resources
 
-		if ((serviceContext.getGroupPermissions() != null) ||
-			(serviceContext.getGuestPermissions() != null)) {
+		ModelPermissions modelPermissions =
+			serviceContext.getModelPermissions();
 
+		String[] groupPermissions = modelPermissions.getActionIds(
```
Comment (dantewang) [inline modules/dxp/apps/portal-reports-engine-console/portal-reports-engine-console-service/src/main/java/com/liferay/portal/reports/engine/console/service/impl/DefinitionLocalServiceImpl.java]: @ChiZelin Can you double check? Original logic in `ServiceContext` checks whether its `_modelPermissions` is null. Is it possible that when you call `getModelPermissions()` the field hasn't been set?

File: portal-impl/test/unit/com/liferay/portal/json/JSONSerializerTest.java (hunk 1 above)
Comment (dantewang) [reply portal-impl/test/unit/com/liferay/portal/json/JSONSerializerTest.java]: See https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/335444/

**Comment by ChiZelin** [comment]
@dantewang force pushed

**Comment by ChiZelin** [comment]
ci:test:sf

**Comment by ChiZelin** [comment]
ci:test

**Comment by dantewang** [comment]
Fetched

**Comment by dantewang** [comment]
Pull request submitted to https://github.com/tinatian/liferay-portal/pull/1604. [See changes here.](https://github.com/dantewang/liferay-portal/compare/a84a9e49fe06~1...a84a9e49fe06)

**Pull request closed without merging**